tests/test_filelock.py

```diff
@@ -675,6 +675,17 @@ def test_singleton_locks_are_distinct_per_lock_file(lock_type: type[BaseFileLock
     assert lock_1 is not lock_2
 
 
+@pytest.mark.parametrize("lock_type", [FileLock, SoftFileLock])
+def test_singleton_locks_must_be_initialized_with_the__same_args(lock_type: type[BaseFileLock], tmp_path: Path) -> None:
+    lock_path = tmp_path / "a"
+    lock = lock_type(str(lock_path), is_singleton=True)  # noqa: F841
```

Comment:
note: noqa since we need the lock to exist for the scope of the test (as a singleton) even though it is "unused".